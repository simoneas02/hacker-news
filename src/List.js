import React from 'react';

import Button from './Button';

const isSearchTerm = searchTerm => item => item.title.toLowerCase().includes(searchTerm.toLowerCase());

const List = ({ list, pattern, onDimiss }) => 
    list.filter( isSearchTerm ( pattern ) ).map( item =>
        <div key={ item.objectId }>
            <h2><a href={ item.url }>{ item.title }</a></h2>
            <p>{ item.author }</p>
            <p>{ item.comments }</p>
            <span>{ item.points }</span>
            <Button onClick={ () => onDimiss( item.objectId ) } >
                Dimiss
            </Button>
        </div>  
)

export default List;