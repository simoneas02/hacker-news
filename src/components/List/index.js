import React from 'react';

import ListItem from '../ListItem';
import { withLoadingIndicator, withNull, withEmpty } from '../../utils';

const List = ({ list, onDimiss }) => 
    <ul>
        {list.map( item => 
            <ListItem key={item.objectID} item={ item } onDimiss={ onDimiss }/>
        )}
    </ul>

const ListWithConditionalRendering = withLoadingIndicator( withNull ( withEmpty ( List) ) );

export default ListWithConditionalRendering;
