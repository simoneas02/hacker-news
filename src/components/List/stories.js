import React from 'react';
import { storiesOf } from '@storybook/react';

import List from '.';

storiesOf('List', module).add("I'm List", () => <h1>Hello List</h1>);
