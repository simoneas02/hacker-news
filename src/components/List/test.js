import React from 'react';
import ReactDOM from 'react-dom';
import renderer from'react-test-renderer';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import List from '.';

Enzyme.configure({ adapter: new Adapter() });

describe('List', () => {
  const props = {
    list: [
      { title: '1', author: '1', num_comments: 1, points: 2, objectID: 'y' },
      { title: '2', author: '2', num_comments: 1, points: 2, objectID: 'z' },
    ],
  };

  it('renders without crashing', () => {
    const div = document.createElement('div');    
    ReactDOM.render(<List { ...props }/>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test('has a valid snapshot', () => {
    const component = renderer.create( <List { ...props }/> );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
})