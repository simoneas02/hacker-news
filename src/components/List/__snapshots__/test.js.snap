// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List has a valid snapshot 1`] = `
<ul>
  <li
    className="list__item"
  >
    <h2>
      <a
        href={undefined}
        target="_blank"
      >
        1
      </a>
    </h2>
    <p>
      1
    </p>
    <p />
    <span>
      2
    </span>
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      Dimiss
    </button>
  </li>
  <li
    className="list__item"
  >
    <h2>
      <a
        href={undefined}
        target="_blank"
      >
        2
      </a>
    </h2>
    <p>
      2
    </p>
    <p />
    <span>
      2
    </span>
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      Dimiss
    </button>
  </li>
</ul>
`;
